### DISCUSSION TITLE
Auto Archive Messages after so long.

### DISCUSSION THREAD

**Discussion by bcordis** [Joomla-Bible-Study/Proclaim#927] [Ideas]
Look at making archive system and then we may even look into a auto trash after so many years. In a way make it zip all that content at that time up so they can store it in a backup system.

**Comment by tomfuller2** [comment]
Super great idea

**Comment by genu** [comment]
Some questions that come to mind:

When we say "archive" do we mean archive a message+media files? If this includes media files, we'll need to generalize this to work around media files stored on third parties, remote servers, etc.

Does the frontend just not show anymore records after the archive threshold?

**Comment by bcordis** [comment]
I this will be a function of Joomla state that the study will be archived but the media will not be touched. Only time we may want to put in a function on the media to allow them to flush old media that no longer available.  This could be a system that checks to see if it is there and puts it in a state with a not of what the server sees and allows them to dele the record or repair the change.

**Comment by bcordis** [comment]
Move to Beta 2

**Comment by bcordis** [comment]
Moving to next version as this did not make it in

**Comment by bcordis** [comment]
With auto-archive will need a plugin to trigger this for us. Almost need a proclaim plugin that is just for these auto tasks?

**Comment by tomfuller2** [comment]
Yes. Makes sense

**Comment by bcordis** [comment]
We also need to work on and display archived versions and searchability.

**Comment by bcordis** [comment]
**Implemented in v10.1** (PR #1045)

The auto-archive feature is now available as a Joomla Scheduled Task:

- **Task Plugin** (`plg_task_proclaim`): New "Archive Old Messages" routine in System > Scheduled Tasks
- **Configurable threshold**: Set the number of days after which published messages are automatically archived
- **Safe operation**: Only archives published messages past the threshold — does not affect already archived or trashed items
- **Joomla native**: Uses Joomla's built-in Task Scheduler, so you can set frequency (daily, weekly, etc.) and execution time

The frontend archived messages display was already implemented earlier in 10.1 with template settings for Published Only / Archived Only / Both, plus optional archive badges.

The remaining items from this discussion (auto-trash after X years, zip/backup content at archive time) are not yet implemented and can be tracked separately.
